### PR TITLE
Split kinds to own modules; unify kind filters with new submodules

### PR DESCRIPTION
**Pull request by Niols**
_built on top of #221 and should only be merged afterwards_

This pull request introduces significant changes to the `Kind` module by splitting it into three sub-modules: `Kind.Base`, `Kind.Version`, and `Kind.Dance`. These changes aim to improve code organisation, enhance visibility, and facilitate integration with new versions of Dune and their `(include_subdirs qualified)` feature.

Additionally, as part of this pull request, the `KindFilters` module has been dismantled, and its functionality has been appropriately distributed among the respective sub-modules (`Kind.Base`, `Kind.Version`, and `Kind.Dance`). This refactoring ensures that the code remains well-organised and follows a modular approach similar to that of the other models.

@R1kM WDYT?

### CODE REVIEW

File: src/common/model/kind/kindVersion.ml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+let%test _ = to_string (32, Waltz) = "32 W"
+let%test _ = to_string (64, Reel) = "64 R"
+let%test _ = to_string (24, Jig) = "24 J"
+let%test _ = to_string (48, Strathspey) = "48 S"
+
+let%test _ = of_string "W 32" = (32, Waltz)
+let%test _ = of_string "64 Reel" = (64, Reel)
+let%test _ = of_string "JIG 24" = (24, Jig)
+let%test _ = of_string "48 sTrathPEY" = (48, Strathspey)
```
Comment (R1kM) [inline src/common/model/kind/kindVersion.ml:32]: How does this test work despite the missing "s" ?

**Review by R1kM — APPROVED** [review]
I have a minor comment, but otherwise it seems like a straightforward, useful refactoring.
Thanks!

File: src/common/model/kind/kindVersion.ml (hunk 1 above)
Comment (Niols) [reply src/common/model/kind/kindVersion.ml:32]: Hah, that's a funny remark and a funny test case. I think we only use the first letter (cf `KindBase.ml` line 31), but that's maybe a bit weak and we'd prefer enforcing either one letter or the whole correct word?

Comment (Niols) [reply src/common/model/kind/kindVersion.ml:32]: Logged in https://github.com/paris-branch/dancelor/issues/227.

**Pull request merged**